app/src/main/java/mozilla/lockbox/presenter/AppRoutePresenter.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.presenter
+
+import android.app.KeyguardManager
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.Navigation
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import io.reactivex.rxkotlin.addTo
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.lockbox.R
+import mozilla.lockbox.action.AppWebPageAction
+import mozilla.lockbox.action.DialogAction
+import mozilla.lockbox.action.RouteAction
+import mozilla.lockbox.action.Setting
+import mozilla.lockbox.action.SettingAction
+import mozilla.lockbox.extensions.view.AlertDialogHelper
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.store.RouteStore
+import mozilla.lockbox.store.SettingStore
+import mozilla.lockbox.view.FingerprintAuthDialogFragment
+
+@ExperimentalCoroutinesApi
+class AppRoutePresenter(
+    private val activity: AppCompatActivity,
+    private val dispatcher: Dispatcher = Dispatcher.shared,
+    private val routeStore: RouteStore = RouteStore.shared,
+    private val settingStore: SettingStore = SettingStore.shared
+) : RoutePresenter(activity, dispatcher, routeStore) {
+
+    override fun onViewReady() {
+        navController = Navigation.findNavController(activity, R.id.fragment_nav_host)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        activity.removeOnBackPressedCallback(backListener)
+        compositeDisposable.clear()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        activity.addOnBackPressedCallback(backListener)
+        routeStore.routes
```

Comment:
Consider making methods to register and unregister listeners and subscriptions, and calling them either from `onResume`/`onPause` or `onViewReady` and `onDestroy`.

This would mean that the `route` method would need to be `protected`/`abstract` in the super class.